modules/home-environment.nix

```
@@ -351,6 +372,10 @@ in
             if [ -n "$__HM_SESS_VARS_SOURCED" ]; then return; fi
             export __HM_SESS_VARS_SOURCED=1
 
+            ${optionalString (hm.os.isLinux cfg.targetOperatingSystem) ''
+              source "${pkgs.nix}/etc/profile.d/nix.sh"
```

I feel like maybe this should also be guarded with a conditional. I use `fish-shell`, and I have a plugin that already deals with sourcing `nix.sh` for me. I don't know a reliable way to do this -- maybe `if [ -n "$NIX_PATH" ]; then`? Curious to hear @rycee's thoughts. Probably doesn't matter.

(Sorry, I keep finding more things to question :P)

---

Also, since you use the dot below, we should stay consistent. (Unless this doesn't work and/or requires `source`?)

```suggestion
              . "${pkgs.nix}/etc/profile.d/nix.sh"
```

---

Sourcing nix.sh multiple times does not have side effects as it will skip the init logic if already sourced.

I should change it to . to be POSIX-compatible, does not change anything besides that

---

My concern was that the `fish-shell` plugin sources `~/.nix-profile/etc/profile.d/nix.sh` directly [1], not just `hm-session-vars.sh`. However, if you're not worried about it, I'm not worried about it.

[1] https://github.com/lilyball/nix-env.fish/blob/master/conf.d/nix-env.fish

---

Oh you are right, the nix.sh does indeed execute the init logic again if called multiple times.. I'm not sure how to fix this. The best way seems to be a new option like `home.skipSourceNixSh` or anything like that..

---

I think that would make things too complicated. At this point, I could just drop the plugin and source `hm-session-vars` manually.

---

Having had a few days to think on it: I still don't think this is necessary. This is now informed by the fact that we use `fenv` to source the `hm-session-vars.sh` script in the `fish` module: https://github.com/rycee/home-manager/blob/a11cf1decd2ae9637672df9ec13f2780111d5256/modules/programs/fish.nix#L347

This means that I can drop the `fish` plugin entirely, and not be required to source `hm-session-vars.sh` manually.

With that, I think this can be marked as resolved (I'd do it myself, but GH only lets collaborators and the PR author to do so).